src/sphinx_book_theme/__init__.py

```diff
@@ -235,7 +235,7 @@ def setup(app: Sphinx):
     app.connect("builder-inited", check_deprecation_keys)
     app.connect("builder-inited", update_sourcename)
     app.connect("builder-inited", update_context_with_repository_info)
-    app.connect("builder-inited", update_general_config)
+    app.connect("config-inited", update_general_config)
```

Comment:
I seem to remember that the reason we did this is because the `config-inited` event actually occurs _before_ the theme is initialized (or something like this) so it wasn't actually being triggered in the way that we thought. Does this make sense and / or are you reasonably certain that we won't run into this issue here?

Reply:
@choldgraf I am certain. Please see my work in [pydata-sphinx-theme #1511](https://github.com/pydata/pydata-sphinx-theme/issues/1511#issuecomment-1767254825). It happens after.

Reply:
Just to confirm - in that issue you mention putting `sphinx_book_theme` in `extensions` as in `extensions = [..., "sphinx_book_theme", ]`. I don't think that's something we want to _require_ for users, we just want it to be listed in `html_theme`[^1]. Does this fix require the book theme to be listed in extensions?

Apologies for the nit-picking here, I've just been burned by the confusing nature of Sphinx events and am inherently distrustful of them 😅

[^1]: For example here it is for this theme's docs:

    https://github.com/executablebooks/sphinx-book-theme/blob/1b6e9eb7496bfedff084a304cd2abe2cb46bd7f5/docs/conf.py#L18-L36

Reply:
> Just to confirm - in that issue you mention putting sphinx_book_theme in extensions as in extensions = [..., "sphinx_book_theme", ]. I don't think that's something we want to require for users, we just want it to be listed in html_theme[1](https://github.com/executablebooks/sphinx-book-theme/pull/770#user-content-fn-1-edb9af0702d9ae3dcab46282b225bd4c). Does this fix require the book theme to be listed in extensions?

Oh wow, that was not clear at all, but see now it was alluded to in [#1660](https://github.com/executablebooks/jupyter-book/issues/1660#issue-1162755546):

> I have narrowed the problem down to the fact that sphinx_book_theme, in addition to being used as the html_theme, is also included as an extension. I'm not too familiar with Sphinx, and am unsure why and if a theme should be included as an extension. However, the [documentation for sphinx_book_theme](https://sphinx-book-theme.readthedocs.io/en/latest/tutorials/get-started.html#) seems to indicate that it is sufficient with html_theme="sphinx_book_theme".

Yes, removing it from `extensions` fixes my problem, but since this has bitten several people downstream:

- [Autosummary and templates not working with versions > 1](https://github.com/executablebooks/sphinx-book-theme/issues/729)
- [jinja2.exceptions.TemplateNotFound: toggle-primary-sidebar.html](https://github.com/sphinx-doc/sphinx/issues/11315)
- [https://github.com/executablebooks/jupyter-book/issues/1660](https://github.com/NREL/floris/issues/635)

1. Could we add a warning note to the docs that this package is a theme only, not an extension?

The error that is thrown points to `pydata-sphinx-theme`, which is misleading and unhelpful.

2. If this package is also to be used as an extension, should `__init__.py::setup()` be modified to trigger on **both** `builder-inited` **and** `config-inited`?

Reply:
> I seem to remember that the reason we did this is because the config-inited event actually occurs before the theme is initialized (or something like this) so it wasn't actually being triggered in the way that we thought.

@choldgraf Switching back to `config-inited` isn't breaking any tests...do you have a regression test in place somewhere?

Reply:
@adam-grant-hendry nice work tracking this down - I've arrived here from your [explanatory comment](https://github.com/executablebooks/sphinx-book-theme/issues/719#issuecomment-1791000570) on the `gemgis` thread.

My attempt to trace the current status and archaeology of this:

* Current status: good news: not many documentation sites have `sphinx-book-theme` in their extensions; so although this has affected a few folks, it appears to me (based on [this approximate GitHub code search](https://github.com/search?q=path%3A%2Fconf.py%24%2F+extensions+%2F%5Cbextensions.*%3D%2F+sphinx_book_theme+%2Fsphinx_book_theme.*%2C%2F+%24&type=code)) that it's relatively limited.  I'll do a bit more searching around and may open a few pull requests as long as I don't find many codebases affected.

* Archaeology: confusing news: my guess is that this was introduced to `gemgis` during a theme migration from `sphinx-rtd-theme`, which intentionally _does_ state -- in contrast -- that it [_should_ be added as an extension](https://sphinx-rtd-theme.readthedocs.io/en/stable/installing.html#how-to-install-and-use-the-theme).  Does it seem unusual that `sphinx-rtd-theme` has to do that?

Reply:
@adam-grant-hendry also a potential puzzler / counterexample: I'm not currently able to replicate this failure when building the `deephyper` documentation (using Sphinx 6.2.1 [from this spec](https://github.com/deephyper/deephyper/blob/92b8bf9d0538b5b39a7e961f2cefedb64fdc9300/setup.py#L86) and with [`sphinx_book_theme` v1.0.1](https://github.com/deephyper/deephyper/blob/92b8bf9d0538b5b39a7e961f2cefedb64fdc9300/setup.py#L87) included in the [Sphinx `extensions` configuration](https://github.com/deephyper/deephyper/blob/92b8bf9d0538b5b39a7e961f2cefedb64fdc9300/docs/conf.py#L79)).

Reply:
Is there a way for us to resolve this in a way that:

- Does not require users to put `sphinx_book_theme` in extensions (because many people don't think to do this in general, and it seems redundant from setting it as a theme)
- Will work the same whether `sphinx_book_theme` is in extensions or not (because some folks will nonetheless use the theme in their extensions list)

Reply:
> Is there a way for us to resolve this

Yes, just add a warning note to your documentation that you should not add it to to `extensions`. I think it is as simple as that.

Reply:
> Does it seem unusual that `sphinx-rtd-theme` has to do that?

No because of the note they add afterwards that explains "Adding this theme as an extension is what enables localization of theme strings in your translated output."

An extension should modify an aspect of the document processing whereas a theme should modify the appearance of the (typically HTML) output.

Reply:
OK makes sense - so does that mean that:

> just add a warning note to your documentation that you should not add it to extensions. I think it is as simple as that

Would then supercede this PR?